lib/Sema/DerivedConformanceRawRepresentable.cpp

```diff
@@ -344,8 +344,9 @@ static bool canSynthesizeRawRepresentable(TypeChecker &tc, Decl *parentDecl,
   auto parentDC = cast<DeclContext>(parentDecl);
   rawType       = parentDC->mapTypeIntoContext(rawType);
 
-  if (!enumDecl->getInherited().empty() &&
-      enumDecl->getInherited().front().isError())
+  auto inherited = enumDecl->getInherited();
+  if (!inherited.empty() && inherited.front().wasValidated() &&
+      inherited.front().isError())
```

Comment:
In fact, I think we've got some tests which test failure to synthesize `RawRepresentable` for `@objc` enums where the raw type is invalid. I can take a look.